src/render/draw_symbol.js

```diff
@@ -168,8 +166,7 @@ function drawTileSymbols(program, programConfiguration, painter, layer, tile, bu
     const tr = painter.transform;
 
     if (isSDF) {
-        const haloWidthProperty = `${isText ? 'text' : 'icon'}-halo-width`;
-        const hasHalo = !layer.isPaintValueFeatureConstant(haloWidthProperty) || layer.paint[haloWidthProperty];
+        const hasHalo = layer.paint.get(`${isText ? 'text' : 'icon'}-halo-width`).constantOr(1) !== 0;
```

Comment:
I was gonna be pretty impressed that Flow could handle this, but it turns out it's just bailing -- adding typo to the property name expression here doesn't produce a flow error.   It does work if the conditional logic is used to select a string literal; i.e., `isText ? 'text-halo-width' : 'icon-halooooo-width'` produces an error, whereas ``${isText ? 'text' : 'icon'}-haloooo-width`` does not.

Reply:
👍 , will switch to `isText ? 'text-halo-width' : 'icon-halo-width'`.